### ISSUE TITLE
Should hiopVectorPar::logBarrier method perform MPI reduction?

### ISSUE THREAD

**Issue by pelesh** [LLNL/hiop#55]
Method `hiopVectorPar::logBarrier` is a reduction, but it sums only over local vector elements and not over all MPI ranks. Is this method intended for use in distributed memory cases?

**Comment by pelesh** [comment]
Other `hiopVectorPar` reduction kernels that perform _local_, but not global (MPI) reduction:
- `fractionToTheBdry`
- `fractionToTheBdry_w_pattern`
-  `isnan`
-  `isinf`
-  `isfinite`

**Comment by cnpetra** [comment]
correct, this is because the calling code performs the interprocess comm for better performance.

I suggest we suffix with `_local` all these functions. Also `linearDampingTerm` falls in the same category

**Comment by pelesh** [comment]
CC: @ashermancinelli

**Comment by pelesh** [comment]
Also, it seems `projectIntoBounds` performs local reduction only, no MPI call.

**Comment by pelesh** [comment]
To summarize - these methods perform only local reduction:
- `logBarrier`
- `linearDampingTerm`
- `projectIntoBounds`
- `fractionToTheBdry`
- `fractionToTheBdry_w_pattern`
-  `isnan`
-  `isinf`
-  `isfinite`

I will document them in GPU portable branch, but will not change function names for now.

**Comment by ashermancinelli** [comment]
Do we ever intend to have globally reducing versions of these functions seeing as the calling code ought to perform the reduction for performance reasons? If not, suffixing with `_local` may be redundant and documenting the responsibilities of the methods ought to be enough.

tldr if we're never going to have globally reducing versions of these functions, I think the names should stay the same and we should just better document/test them.

**Comment by cnpetra** [comment]
likely not, but I still think they should be suffixed. It's going to be just a hell more clear that there is no communic...

**Comment by ashermancinelli** [comment]
SGTM

**Comment by cnpetra** [comment]
on my to do list

**Comment by cnpetra** [comment]
closing ... commit a0074ae addresses this.
